catalog/spatial/wfs/spatial-wfs-featuretransformer-handlebars/src/main/java/org/codice/ddf/spatial/ogc/wfs/transformer/handlebars/HandlebarsWfsFeatureTransformer.java

```diff
@@ -0,0 +1,673 @@
+    String wkt = null;
+    Gml3ToWkt gml3ToWkt = new Gml3ToWktImpl(new org.geotools.gml2.GMLConfiguration());
+    try {
+      wkt = gml3ToWkt.convert(geometry);
+    } catch (Exception e) {
+      LOGGER.debug("Error converting gml to wkt using gml2 configuration. GML: {}.", geometry, e);
+    }
+    return wkt;
+  }
+
+  private boolean isPrefixBound(StartElement startElement, String prefix) {
+    for (Iterator i = startElement.getNamespaces(); i.hasNext(); ) {
+      Namespace namespace = (Namespace) i.next();
+
+      if (namespace.getPrefix().equals(prefix)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  private void mapNamespaces(StartElement startElement, Map<String, String> map) {
+
+    for (Iterator i = startElement.getNamespaces(); i.hasNext(); ) {
+      Namespace namespace = (Namespace) i.next();
+      map.put(namespace.getPrefix(), namespace.getNamespaceURI());
+    }
+  }
+
+  private boolean isStateValid(InputStream inputStream, WfsMetadata metadata) {
+    if (inputStream == null) {
+      LOGGER.debug("Received a null input stream.");
+      return false;
+    }
+
+    if (metadata == null) {
+      LOGGER.debug("Received a null WfsMetadata object.");
+      return false;
+    }
+
+    if (StringUtils.isBlank(featureType)) {
+      LOGGER.debug("Feature type must contain a value: {}", featureType);
+      return false;
+    }
+
+    if (CollectionUtils.isEmpty(mappingEntries.values())) {
+      LOGGER.debug("There are no mappings for feature type: {}", featureType);
+      return false;
+    }
+
+    return true;
+  }
+
+  private void addAttributeMapping(String attributeName, String featureName, String templateText) {
+    LOGGER.trace(
+        "Adding attribute mapping from: {} to: {} using: {}",
+        attributeName,
+        featureName,
+        templateText);
+    mappingEntries.put(
+        featureName, new FeatureAttributeEntry(attributeName, featureName, templateText));
+  }
+
+  public String getDataUnit() {
+    return dataUnit;
+  }
+
+  public void setDataUnit(String unit) {
+    LOGGER.trace("Setting data unit to: {}", unit);
+    dataUnit = unit;
+  }
+
+  public void setFeatureType(String featureType) {
+    LOGGER.trace("Setting feature type to: {}", featureType);
+    this.featureType = featureType;
+    featureTypeQName = QName.valueOf(featureType);
+  }
+
+  public void setMetacardTypeRegistry(WfsMetacardTypeRegistry metacardTypeRegistry) {
+    this.metacardTypeRegistry = metacardTypeRegistry;
+  }
+
+  /**
+   * Sets a list of attribute mappings from a list of JSON strings.
+   *
+   * @param attributeMappingsList - a list of JSON-formatted `FeatureAttributeEntry` objects.
+   */
+  public void setAttributeMappings(/*@Nullable*/ List<String> attributeMappingsList) {
+    LOGGER.trace("Setting attribute mappings to: {}", attributeMappingsList);
+    if (attributeMappingsList != null) {
+      mappingEntries.clear();
+      attributeMappingsList
```

Comment:
Can we extract some of this into separate methods for readability?  Can the filters be condensed into 1?

Reply:
refactored for readability